zebrad/src/components/mempool/storage/tests/prop.rs

```diff
@@ -0,0 +1,249 @@
+use std::fmt::Debug;
+
+use proptest::prelude::*;
+use proptest_derive::Arbitrary;
+
+use zebra_chain::{
+    at_least_one, orchard,
+    primitives::Groth16Proof,
+    sapling,
+    transaction::{self, Transaction, UnminedTx},
+    transparent, LedgerState,
+};
+
+use super::super::{MempoolError, Storage};
+
+proptest! {
```

Comment:
Performance / Coverage Tradeoff: How long do these new tests run for?
We usually try to keep individual tests under 10-20 seconds.

Since there are 6 test variants, we don't want to go much below 32 cases, to make sure we cover all of them.
(The first few proptest cases can just be seeds.)

But the default is 256 cases, so we have some room to tweak performance if we need to.

Here's how we usually do that:
https://github.com/ZcashFoundation/zebra/blob/bcd5f2c50d83dd4fabbbbc1704b10edacf0ea9c8/zebra-network/src/peer_set/candidate_set/tests/prop.rs#L87-L91

Reply:
It's currently taking 6 seconds on my local computer. Should I reduce the number of test cases?

(Random idea: it would be cool if there was a `proptest` feature that automatically limits tests to 1 second by default, and would somehow persist which tests are slow to vary the inputs more. This could then be disabled on CI, either for each pull request or in a new "full test" job that runs daily.)

Reply:
6 seconds is fine.